client/app/bundles/course/assessment/pages/AssessmentStatistics/AnswerDisplay/AllAttemptsDisplay.tsx

```diff
@@ -0,0 +1,155 @@
+import { FC, useState } from 'react';
+import { defineMessages } from 'react-intl';
+import { Slider, Typography } from '@mui/material';
+import { QuestionType } from 'types/course/assessment/question';
+import {
+  AllAnswerDetails,
+  QuestionDetails,
+} from 'types/course/statistics/assessmentStatistics';
+
+import { workflowStates } from 'course/assessment/submission/constants';
+import Accordion from 'lib/components/core/layouts/Accordion';
+import Link from 'lib/components/core/Link';
+import useTranslation from 'lib/hooks/useTranslation';
+import { formatLongDateTime } from 'lib/moment';
+
+import AnswerDetails from '../AnswerDetails/AnswerDetails';
+
+interface Props {
+  allAnswers: AllAnswerDetails<keyof typeof QuestionType>[];
+  question: QuestionDetails<keyof typeof QuestionType>;
+  questionNumber: number;
+  submissionEditUrl: string;
+}
+
+const translations = defineMessages({
+  questionTitle: {
+    id: 'course.assessment.statistics.questionTitle',
+    defaultMessage: 'Question {index}',
+  },
+  pastAnswerTitle: {
+    id: 'course.assessment.statistics.pastAnswerTitle',
+    defaultMessage: 'Submitted At: {submittedAt}',
+  },
+  mostRecentAnswer: {
+    id: 'course.assessment.statistics.mostRecentAnswer',
+    defaultMessage: 'Current Answer (Attempting)',
+  },
+  submissionPage: {
+    id: 'course.assessment.statistics.submissionPage',
+    defaultMessage: 'Go to Answer Page',
+  },
+});
+
+// only used as a dummy buffer time if the submission's state is attempting
+// this is due to the time record for attempting answer being lower than any other
+// states' answer, while it's actually the latest version.
+const BUFFER_TIME = 100;
+
+const AllAttemptsDisplay: FC<Props> = (props) => {
+  const { allAnswers, question, questionNumber, submissionEditUrl } = props;
+
+  const { t } = useTranslation();
+
+  let currentAnswer = allAnswers.find((answer) => answer.currentAnswer);
+  const sortedAnswers = allAnswers.filter((answer) => !answer.currentAnswer);
+
+  if (
+    sortedAnswers.length > 0 &&
+    currentAnswer?.workflowState === workflowStates.Attempting
+  ) {
+    currentAnswer = {
+      ...currentAnswer,
+      createdAt: new Date(
+        sortedAnswers[sortedAnswers.length - 1].createdAt.getTime() +
+          BUFFER_TIME,
+      ),
+    };
+  }
+
+  sortedAnswers.push(currentAnswer!);
+
+  const answerSubmittedTimes = sortedAnswers.map((answer, idx) => {
+    return {
+      value: idx,
+      label:
+        idx === 0 || idx === sortedAnswers.length - 1
+          ? formatLongDateTime(answer.createdAt)
+          : '',
```

Comment:
Why do the first AND last have `''` label?

Reply:
no, it's the other way around.. only the first and last have created_at to be displayed inside the slider
otherwise, all the labels beneath the slider will be interleaving to each other